plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -4230,4 +4223,26 @@ public boolean isSecureMode(String bootMode) {
 
         return false;
     }
+
+    private void setCpuTopology(CpuModeDef cmd, int vcpus, Map<String, String> details) {
```

Comment:
@ustcweizhou possible to pass _coresPerSocket_ here, instead complete VM _details_ here ?

Reply:
@sureshanaparti 
I was thinking about it. at the end I decided to pass vm details based on two reasons
(1) setCpuTopology is a complete function now. if pass coresPerSocket , we need to get coresPerSocket out of the method.
(2) it is better if add more cpu settings , for example the pr #4178

Reply:
> @sureshanaparti
> I was thinking about it. at the end I decided to pass vm details based on two reasons
> (1) setCpuTopology is a complete function now. if pass coresPerSocket , we need to get coresPerSocket out of the method.
> (2) it is better if add more cpu settings , for example the pr #4178

Ok @weizhouapache so going forward, cpu hyperthreading detail mentioned in PR #4178 will be added to this method, right?

Reply:
@sureshanaparti yes. but I need to discuss with @div8cn about some details. it will be an improvement in 4.16

Reply:
> @sureshanaparti yes. but I need to discuss with @div8cn about some details. it will be an improvement in 4.16

OK, Pls update details here. As this PR is targeted for 4.14.1.0, the proposed changes with cpu hyperthreading detail will be targeted for 4.16, Right?

Reply:
@sureshanaparti yes, this is a bug which should be fixed in 4.14 and 4.15.
the other pr is an improvement. it will not impact this pr.